### ISSUE TITLE
Resizing is kinda broken cuz im dumb af

### ISSUE THREAD

**Issue by briarsychung** [briarsychung/tech_guy#55]
It seems like you can't resize two WrappedCanvas at once; I have made it so that there are two of them each with a resize event listener on window (there needs to be a re-sizable cam but also a resize-able desk), but when on one screen and resizing, the other screen's canvas becomes blank. (On a related note, how do you initialize the resize when the page loads?)

I think a big problem here is that students-roam is a demo file, perhaps if there was a module for the camera it would be easier to work with.

Additionally, it seems like the camera canvas is resized in students roam AND in canvas stuff so I'm unsure why it needs to be resized twice.

**Comment by SheepTester** [comment]
Yeah the resize thing needs to be more intellectual because it won't resize when hidden

I think I complained about this somewhere but I forgot where

A quick solution would just be to use `visibility: hidden;` I think, or `opacity: 0; pointer-events: none;`, so that the browser calculates the size of the canvas wrapper when resizing even if it's hidden

**Comment by briarsychung** [comment]
Yeah there was some "TODO: broken" or something. I'll try to standardize desk.js and hopefully students-roam.js can follow suit.

**Comment by briarsychung** [comment]
Also @SheepTester how do you initialize the resize when the page loads. Also using visibility doesn't work because the camera and office views will stack (they are flex elements)

**Comment by briarsychung** [comment]
Another update: the issue is that the entirety of `students-roam.js` is called in `index.html` via `main()`, wherein it just runs forever. My solution was going to be to update the sizes of all resizing elements when the windows resizes (already in the code), AND when the player switches from office to camera or vice versa, but I can't do that because I can't access the wrapped canvas that is the camera view in `index`, or anything else for that matter.

**Comment by SheepTester** [comment]
I probably should just redo students-roam since it's just a demo
